### PR TITLE
Release publish NPM packages

### PR DESCRIPTION
**Pull request by GImbrailo**
This PR adds an additional dependency on the `build-and-publish-npm-packages` job to be successful. There are also three extra steps in the `build-and-publish-npm-packages` job to 
* publish packages to npmjs.com with a `pr-#` tag on `pull_request`
* publish the packages to npmjs.com with the `next` tag if there is a suffix after a git tag version number
* publish the packages to npmjs.com with the `latest` tag if there is no suffix after the git tag version number

The PR comment has also been updated to remove the `--registry` argument and value as well as show the user what version of the package to install after a commit.

The binary name has been changed to `taq` and this PR renames all references and binary names from `taqueria` to `taq`

**Testing Plan**
Please see [release v0.0.1-test.29](https://github.com/ecadlabs/taqueria/releases/tag/untagged-7355f01a3e5af073bec0) for assets
image

Please see the comment on this PR for proof of the package version being commented on the ticket
image

Please see the [npmjs.com registry](https://www.npmjs.com/package/@taqueria/plugin-ligo) for proof of uploaded packages and tags
image

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -31,15 +31,15 @@ jobs:
...
-           taqueria_bin: "taqueria"
+           taqueria_bin: "taq"
+        #  - os: windows-latest
+        #    os_short: windows
+        #    deno_target: "x86_64-pc-windows-msvc"
+        #    taqueria_bin: "taq.exe"
+        #  - os: macOS-latest
+        #    os_short: macos
+        #    deno_target: "x86_64-apple-darwin"
+        #    taqueria_bin: "taq"
```
Comment (danielelisi) [inline .github/workflows/main.yml:42]: The repo is now public so It's okay to include Windows and MacOS jobs

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -113,74 +112,57 @@ jobs:
...
+          npm --no-git-tag-version version "0.0.0-pr-${{ github.event.number }}-${GITHUB_SHA::8}"
+          npm publish --tag "pr-${{ github.event.number }}" --access public
       
-      - name: Publish if new version of package 
+      - name: Publish for prerelease
         env: 
-          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        if: steps.get-npm-package-version.outputs.version < steps.get-local-package-version.outputs.version
-        run: npm publish 
+          NODE_AUTH_TOKEN: ${{ secrets.NPMJS_PAT }}
```
Comment (danielelisi) [inline .github/workflows/main.yml:140]: You can make this env variable available for the entire `job` if you intend to use it in more than one step

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -203,13 +185,23 @@ jobs:
...
-              | --------- |
-              | npm install ${{ env.SCOPE }}/${ligo.name}@${ligo.version} --registry=${{ env.GH_REGISTRY }}|
-              | npm install ${{ env.SCOPE }}/${smartpy.name}@${smartpy.version} --registry=${{ env.GH_REGISTRY }}|
-              | npm install ${{ env.SCOPE }}/${taquito.name}@${taquito.version} --registry=${{ env.GH_REGISTRY }}|
-              | npm install ${{ env.SCOPE }}/${sdk.name}@${sdk.version} --registry=${{ env.GH_REGISTRY }}|
+              | ------------------------------------- |
+              | npm install @taqueria/${ligo.name}@${ligo.version}-${{ github.event.number }}-${git_short_sha}|
+              | npm install @taqueria/${smartpy.name}@${smartpy.version}-${{ github.event.number }}-${git_short_sha}|
+              | npm install @taqueria/${taquito.name}@${taquito.version}-${{ github.event.number }}-${git_short_sha}|
+              | npm install @taqueria/${sdk.name}@${sdk.version}-${{ github.event.number }}-${git_short_sha}|
```
Comment (danielelisi) [inline .github/workflows/main.yml:204]: Why are you hardcoding this? referencing a common variable is usually preferred

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -222,21 +214,35 @@ jobs:
...
       - name: Download binaries 
         id: download-binaries 
         run: |
           mkdir release && cd release
-          wget ${{ needs.build-and-test.outputs.public-url-ubuntu }} -O taqueria.${GITHUB_REF/refs\/tags\//}-linux
-          wget ${{ needs.build-and-test.outputs.public-url-windows }} -O taqueria.${GITHUB_REF/refs\/tags\//}-windows.exe
-          wget ${{ needs.build-and-test.outputs.public-url-macos }} -O taqueria.${GITHUB_REF/refs\/tags\//}-macos
+          wget ${{ needs.build-binaries.outputs.public-url-ubuntu }} -O taq.${GITHUB_REF/refs\/tags\//}-linux
+        # wget ${{ needs.build-binaries.outputs.public-url-windows }} -O taq.${GITHUB_REF/refs\/tags\//}-windows.exe
+        # wget ${{ needs.build-binaries.outputs.public-url-macos }} -O taq.${GITHUB_REF/refs\/tags\//}-macos
```
Comment (danielelisi) [inline .github/workflows/main.yml:240]: You can re-include the Windows and MacOS steps now that the repo is public

File: .github/workflows/main.yml (hunk 1 above)
Comment (GImbrailo) [reply .github/workflows/main.yml]: I have added these again

**Review by alexzbusko — APPROVED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (GImbrailo) [reply .github/workflows/main.yml]: Thanks for pointing that out, I reverted this change

File: .github/workflows/main.yml (hunk 2 above)
Comment (GImbrailo) [reply .github/workflows/main.yml]: I changed the env variable to the `job` scope

File: .github/workflows/main.yml (hunk 4 above)
Comment (GImbrailo) [reply .github/workflows/main.yml]: These have been added once again

**Review by danielelisi — APPROVED** [review]
LGTM 👍

**Pull request merged**